### ISSUE TITLE
Disable video with fully dimmed background

### ISSUE THREAD

**Issue by smoogipoo** [ppy/osu#13294]
Quite an important one since video decoding (especially on the CPU) can be quite expensive, and I don't think the decoding loop is paused while video is faded out/etc so this might also require framework changes.

---

i think it will be better if video from beatmap will be disabled if backround dim is equal to 100%
noticed that video is already playing while dim is 100% that caused fps drops

_Originally posted by @InsanityPP in https://github.com/ppy/osu/discussions/13289_

**Comment by peppy** [comment]
Definitely should be pausing the decoding if the drawable isn't updating.

For the disabling part, I'd want to see this done in an elegant way, since those `UserDimContainer` classes have previously gotten pretty hard to follow with all the different dim/disable types. Maybe an `AggregateBindable` or something.

**Comment by integerrr** [comment]
should this be extended to disable storyboard with full dimmed bg as well?

**Comment by peppy** [comment]
Yes. They should both be in the same drawable hierarchy anyway.

**Comment by LumpBloom7** [comment]
The `VideoDecoder` is written in a way that as long as `Video` doesn't retrieve the decoded frames from the queue, no new frames will be decoded, and the decoding task goes into a spin cycle. (The queue threshold is set to 3)

When user dim is set to 100%, the `Video` drawable isn't present even if storyboards are enabled, and therefore doesn't retrieve the decoded frames from the decoder. (Except of course during break time, where the dim is slightly decreased)

This issue could be closed if the empty spin cycle isn't a problem.

**Comment by peppy** [comment]
Confirmed and added test coverage at https://github.com/ppy/osu-framework/pull/4572.
